wagtail/wagtailsnippets/templates/wagtailsnippets/snippets/type_index.html

```diff
@@ -9,8 +9,10 @@
                 <h1 class="icon icon-snippet">{% blocktrans with snippet_type_name_plural=snippet_type_name_plural|capfirst %}Snippets <span>{{ snippet_type_name_plural }}</span>{% endblocktrans %}</h1>
             </div>
             <div class="right col3">
-                <a href="{% url 'wagtailsnippets:add' content_type.app_label content_type.model %}" class="button bicolor icon icon-plus">{% blocktrans %}Add {{ snippet_type_name }}{% endblocktrans %}</a>
-                {# TODO: figure out a way of saying "Add a/an [foo]" #}
+                {% if can_add_snippet %}
+                    <a href="{% url 'wagtailsnippets:add' content_type.app_label content_type.model %}" class="button bicolor icon icon-plus">{% blocktrans %}Add {{ snippet_type_name }}{% endblocktrans %}</a>
+                    {# TODO: figure out a way of saying "Add a/an [foo]" #}
```

Comment:
Hmm, possibly... we'd have to work out a way to only apply this on English text, and preserve the old behaviour for other languages. (I'm assuming this is not solvable for all languages in the general case :-) )

Just installed it to have a play, and it looks like it needs updating for new Django versions - currently failing with `ImportError: cannot import name 'force_unicode'`.